python/lsst/ts/m2gui/model.py

```diff
@@ -269,3 +269,46 @@ def report_config(self, **kwargs):
         self.signal_config.config.emit(config)
 
         return config
+
+    def save_position(self):
+        """Save the rigid body position."""
+        self.log.info("Save the rigid body position.")
+
+    def go_to_position(self, x, y, z, rx, ry, rz):
+        """Go to the position.
+
+        Parameters
+        ----------
+        x : `float`
+            Position x in mm.
+        y : `float`
+            Position y in mm.
+        z : `float`
+            Position z in mm.
+        rx : `float`
+            Rotation x in urad.
+        ry : `float`
+            Rotation y in urad.
+        rz : `float`
+            Rotation z in urad.
+        """
+
+        if self.is_enabled_and_closed_loop_control():
```

Comment:
Ultimately your code closer to hardware will decide if mirror can be moved. My preference would be to use is_enabled_and_closed_loop_control inside GUI which has "Move mirror" button to disable/enable the button. Of course you need to handle errors then. Probably best is to raise an exception if Move mirror command cannot be executed, so the GUI part can decided what to do (show a dialog, reporting problem..). Your current design hides failure from GUI - you don't expect GUI to parse log to learn what was wrong. Please modify that to raise e.g. RuntimeException if mirror cannot move, and catch that in _callback_go_to_position

Reply:
The error handling will be implemented in DM-34861. Thanks!